packages/bundlers/default/src/DefaultBundler.js

```diff
@@ -180,6 +210,7 @@ export default (new Bundler({
             nullthrows(bundleRoots.get(existingBundle)).push(asset);
             bundlesByEntryAsset.set(asset, existingBundle);
             bundleGraph.createAssetReference(dependency, asset);
+            siblingBundles.add(existingBundle);
```

Comment:
Second problem in React Spectrum (couldn't find a test case because this is likely caused by Devon's custom plugins, not sure if this is the correct solution)

https://github.com/adobe/react-spectrum/blob/parcel-4884-repro-1/run.sh.png
https://github.com/adobe/react-spectrum/blob/parcel-4884-repro-1/run.sh.png
I think it's somehow caused by the `isIsolated:true` `client` dependency of `Checkbox.mdx`